Archives/ClmFile.cpp

```diff
@@ -366,7 +321,7 @@ namespace Archives
 
 	// Compares wave format structures in the waveFormats container
 	// Returns true if they are all the same and false otherwise.
-	bool ClmFile::CompareWaveFormats(const std::vector<WaveFormatEx>& waveFormats)
+	bool ClmFile::CompareWaveFormats(const std::vector<WaveFormatEx>& waveFormats) const
```

Comment:
Hmm, maybe consider making this function `static` rather than `const`. It doesn't modify any member variables, and hence const is appropriate, but it also doesn't even read any member variables either, so `static` would be even more appropriate. Also, to eliminate the need for the template files, we should be making all the write functions static so they can be called without an existing class instance. It was wrong of me to make the file writing function virtual.

Reply:
I was thinking of eventually separating ClmFile into ClmWriter and ClmReader. This could be ClmPacker and ClmUnpacker as well. I hadn't thought about using static functions for writing the CLM File. Would you prefer to see the ClmFile class remain with both packing and unpacking available? If we want to separate into 2 classes, it might be best to wait until later to separate them.

I'll go ahead and mark this function as static in any case.

Reply:
Yeah, trying to separate them seems like a lot more work, and I'm not sure it's entirely justified. I think we should move the packing to static methods while keeping it part of the same class. That should be easier to do, and it should eliminate the need for template files.

The framework I use for evaluating if something should be a static method, or a class, is the time frame over which operations are performed. If it's a single uninterruptible operation, such as pack, a static method makes sense. If instead you have an object representing a resource, such as a CLM file, and operations can be performed over time, such as periodically opening or reading from a new stream, then an object makes more sense. The CLM file is not typically opened, parsed, and loaded all into memory at once as some collection of wave objects, where upon the CLM file is then close and forgotten about. A CLM object for reading makes sense as you can perform multiple read operations over time, rather than a single one time operation.